.gitlab/dev_container_deploy/docker_linux.yml

```diff
@@ -225,12 +225,15 @@ single_machine_performance-nightly-amd64-a7:
   stage: dev_container_deploy
   rules:
     !reference [.on_deploy_nightly_repo_branch_a7]
+  only:
+    variables:
+      - $CI_COMMIT_BRANCH == "main"
```

Comment:
According to [the documentation](https://docs.gitlab.com/ee/ci/yaml/#rules), `only` is deprecated and cannot work alongside `rules`:
> `rules` replaces `only/except` and they can’t be used together in the same job. If you configure one job to use both keywords, the GitLab returns a `key may not be used with rules` error.